02-basiccpp/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+cmake_minimum_required(VERSION 3.16.0 FATAL_ERROR)
+project(workshop-02)
+
+# ------------------------------------------------- #
+# CONFIGURE CLANG                                   #
+# ------------------------------------------------- #
+
+
+set(CMAKE_CXX_COMPILER "clang++")
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -Wall -Wextra -Wpedantic")
+set(CMAKE_CXX_LINKER_FLAGS "${CMAKE_CXX_LINKER_FLAGS} -stdlib=libc++ -lpthread -Wall -Wextra -Wpedantic")
+
+message("compile flags: ${CMAKE_CXX_FLAGS}")
+message("link flags:    ${CMAKE_CXX_LINKER_FLAGS}")
+
+# ------------------------------------------------- #
+# GOOGLE TEST                                       #
+# ------------------------------------------------- #
+
+include(ExternalProject)
+
+ExternalProject_Add(
```

Comment:
There's a much better way of doing this (which doesn't require building GTest every time): https://google.github.io/googletest/quickstart-cmake.html

I've done this in the main repo already.